pkg/experiment/symbolizer/symbolizer.go

```diff
@@ -0,0 +1,753 @@
+		return fmt.Errorf("open lidia file: %w", err)
+	}
+	defer table.Close()
+
+	return s.symbolizeWithTable(ctx, table, ei, req)
+}
+
+func (s *ProfileSymbolizer) storeLidiaTableInCache(data []byte, buildID string, cacheStart time.Time) error {
+	// Process the ELF data
+	lidiaData, ei, err := s.processELFData(data)
+	if err != nil {
+		return err
+	}
+
+	// Store both the processed Lidia table data and the BinaryLayout in the cache
+	entry := LidiaTableCacheEntry{
+		Data: lidiaData,
+		EI:   ei,
+	}
+
+	success := s.lidiaTableCache.Set(buildID, entry, int64(len(lidiaData)))
+	s.metrics.cacheOperations.WithLabelValues("debuginfo_cache", "set", StatusSuccess).Observe(time.Since(cacheStart).Seconds())
+	if !success {
+		level.Warn(s.logger).Log("msg", "Failed to store debug info in cache", "buildID", buildID, "size", len(lidiaData))
+	}
+	s.metrics.cacheSizeBytes.WithLabelValues("debuginfo_cache").Set(float64(len(lidiaData)))
+
+	return nil
+}
+
+func (s *ProfileSymbolizer) processELFData(data []byte) (lidiaData []byte, ei *BinaryLayout, err error) {
+	// Create a reader from the data
+	reader, err := detectCompression(bytes.NewReader(data))
+	if err != nil {
+		return nil, nil, fmt.Errorf("detect compression: %w", err)
+	}
+
+	// Parse the ELF file
+	sr := io.NewSectionReader(reader, 0, 1<<63-1)
+	elfFile, err := elf.NewFile(sr)
+	if err != nil {
+		return nil, nil, fmt.Errorf("parse ELF file: %w", err)
+	}
+	defer elfFile.Close()
+
+	// Get executable info for address normalization
+	ei, err = ExecutableInfoFromELF(elfFile)
+	if err != nil {
+		return nil, nil, fmt.Errorf("executable info from ELF: %w", err)
+	}
+
+	// Create an in-memory buffer for the lidia format
+	initialSize := len(data) * 2 // A simple heuristic: twice the compressed size
+	memBuffer := newMemoryBuffer(initialSize)
+
+	// Create lidia file from ELF directly in memory
+	err = lidia.CreateLidiaFromELF(elfFile, memBuffer, lidia.WithCRC(), lidia.WithFiles(), lidia.WithLines())
+	if err != nil {
+		return nil, nil, fmt.Errorf("create lidia file: %w", err)
+	}
+
+	return memBuffer.Bytes(), ei, nil
+}
+
+// updateSymbolCache updates the symbol cache with newly symbolized addresses
+func (s *ProfileSymbolizer) updateSymbolCache(req *Request) {
+	if s.symbolCache == nil {
+		return
+	}
+
+	for _, loc := range req.Locations {
+		if len(loc.Lines) > 0 {
+			cacheKey := s.createSymbolCacheKey(req.BuildID, loc.Address)
+			cacheStart := time.Now()
+			s.symbolCache.Add(cacheKey, loc.Lines)
+			s.metrics.cacheOperations.WithLabelValues("symbol_cache", "set", StatusSuccess).Observe(time.Since(cacheStart).Seconds())
+		}
+	}
+}
+
+func (s *ProfileSymbolizer) createNotFoundSymbols(binaryName string, loc *Location, addr uint64) []lidia.SourceInfoFrame {
+	prefix := "unknown"
+	if binaryName != "" {
+		prefix = binaryName
+	}
+
+	return []lidia.SourceInfoFrame{{
+		FunctionName: fmt.Sprintf("%s!0x%x", prefix, loc.Address),
+		FilePath:     fmt.Sprintf("mapped_0x%x", addr),
```

Comment:
I think we can leave FilePath empty. I'ts not visible anywhere and not useful. Mapping file path is enough.